discretisedfield/field.py

```diff
@@ -3059,112 +3061,98 @@ def _fromovf(cls, filename):
-        # extracted here.
-        if 'valuedim' not in mdatadict.keys():
-            mdatadict['valuedim'] = len(datalines) / len(mesh)
-
-        r_tuple = (*tuple(reversed(mesh.n)), int(mdatadict['valuedim']))
-        t_tuple = (*tuple(reversed(range(3))), 3)
-
-        return cls(mesh, dim=int(mdatadict['valuedim']),
-                   value=datalines.reshape(r_tuple).transpose(t_tuple))
+                information = line.split()
+                if len(information) > 1:
+                    key = information[1][:-1]  # remove trailing colon
+                    if key in mdata_float:
+                        header[key] = float(information[-1])
+                    elif key in mdata_int:
+                        header[key] = int(information[-1])
+                    elif key in mdata_str:
+                        header[key] = information[-1]
+                    elif key in mdata_list:
+                        header[key] = information[-3:]
+
+            if header['ovf_version'] == '1':
+                # valuedim is fixed to 3 and not in the header
+                header['valuedim'] = 3
+
+            if mode == 'Binary':
+                if header['ovf_version'] == '1':
+                    endian = '>'  # ovf1 uses big-endian
+                else:
+                    endian = '<'  # ovf2 uses little-endian
+
+            # >>> MESH <<<
+            p1 = (header[key] for key in ['xmin', 'ymin', 'zmin'])
+            p2 = (header[key] for key in ['xmax', 'ymax', 'zmax'])
+            cell = (header[key] for key in ['xstepsize', 'ystepsize',
+                                            'zstepsize'])
+            mesh = df.Mesh(region=df.Region(p1=p1, p2=p2), cell=cell)
+
+            # >>> READ DATA <<<
+            if mode == 'Binary':
+                decoder = struct.Struct(
+                    f'{endian}{"d" if nbytes == 8 else "f"}')
+
+                test_data = decoder.unpack(f.read(nbytes))[0]
+                check = {4: 1234567.0, 8: 123456789012345.0}
+                if nbytes not in (4, 8) or test_data != check[nbytes]:
+                    raise ValueError(  # pragma: no cover
+                        f'Cannot read file {filename}. The file seems to be in'
+                        ' binary format ({nbytes} bytes) but the check value'
+                        f' is not correct: Expected {check[nbytes]}, got'
+                        f' {check}.')
+
+                data = f.read()
+
+                # length of b'# End: Data Binary X\n# End: Segement\n' (37)
+                # plus additional safety
+                footer_length_estimate = 45
+                end = data.find(b'# End: Data Binary', footer_length_estimate)
+
+                data = data[:end].strip()  # trailing newline character in ovf
+                bdata = decoder.iter_unpack(data)
+                dtype = np.float64 if nbytes == 8 else np.float32
+                array = np.fromiter((e[0] for e in bdata), dtype=dtype)
```

Comment:
very nice :)

Reply:
I do not know if this could help: https://numpy.org/doc/stable/reference/generated/numpy.fromfile.html

Reply:
Switching to `np.fromfile` for binary data gave an additional speedup of ~8.